### ISSUE TITLE
Unable to set MaxMetaspaceSize

### ISSUE THREAD

**Issue by redbaron** [sbt/sbt#6012]
## steps

```
$ cat .jvmopts
-XX:MaxMetaspaceSize=1024M
```

## problem

sbt launcher still removes MaxMetaspaceSize from passed JVM opts if it doesn't include any other memory option.

## expectation

`-XX:MaxMetaspaceSize=1024M` is passed in.

## note

This works:

```
$ cat .jvmopts
-Xmx1G
-XX:MaxMetaspaceSize=1024M
```

**Comment by eatkins** [comment]
Just out of curiosity, what is your use case for setting the MaxMetaspaceSize? In my experience, limiting it can cause problems. What problem is solved by setting the limit? A lot of builds seem to have switched from setting MaxPermGen to MaxMetaspaceSize after upgrading to java 8. In older versions of java, it was necessary to _increase_ the permgen size because the default was too low for sbt use cases. By default, metaspace is unlimited so setting the MaxMetaspaceSize parameter adds a limit where there previously was none, which mainly just increases the likelihood that sbt crashes with an OOM.

**Comment by redbaron** [comment]
My builds were failing on CI I don't have full control of, which currently seems to set it unconditionally. I was trying to replicate problem locally and found this issue.

**Comment by eatkins** [comment]
Makes sense. One thing you can do in the meantime is bypass the sbt launcher script and call `java -XX:MaxMetaspaceSize=1024m -jar sbt-launch.jar`

**Comment by eed3si9n** [comment]
```bash
$ cat .jvmopts
-XX:MaxMetaspaceSize=1024M

$ sbt -v
[sbt_options] declare -a sbt_options=([0]="-Xms2048M" [1]="-Xmx2048M" [2]="-Xss2M")
[process_args] java_version = '8'
# Executing command line:
java
-Dfile.encoding=UTF-8
-XX:MaxMetaspaceSize=1024M
-Dsbt.script=/Users/xxx/.sdkman/candidates/sbt/current/bin/sbt
-Xms2048M
-Xmx2048M
-Xss2M
-jar
/Users/xxx/.sdkman/candidates/sbt/1.12.0/bin/sbt-launch.jar
```

This seems to work now.
